### DISCUSSION TITLE
Kod stili

### DISCUSSION THREAD

**Discussion by haxala1r** [melikechan/effl-comp-olympiads#4] [General]
Az önce açtığım PR'ımı hazırlarken bir sorunla karşılaştım.

Şuanki haliyle, bu repoda paylaşılan kod dosyalarının izlemesi gereken herhangi bir stil bulunmuyor. Bu da dosyaların genel olarak kötü gözükmesine sebep oluyor. Bana kalırsa bu tür soruların (boşluk mu yoksa tab mı kullanılmalı? peki ya dosya sonunda yeni satır? değişkenler nasıl adlandırılmalı? vb. vb.) cevaplandırıldığı bir dosya (belki bir "CONTRIBUTING.md" ?) oluşturulmalı.

Sizin bu konudaki fikirleriniz neler?

**Comment by melikechan** [comment]
Bence de iyi olur çünkü doğruyu söylemek gerekirse VSCode'da tam bir tab stili ayarlayamamıştım. Dosyalara da fazla bakamıyordum doğrusu. Açıkçası C ile fazla ilgilenmediğimden dolayı evrensel kod stilini bilmiyorum fakat onun için bir contributing.md oluşturabiliriz.

**Reply by haxala1r** [reply]
Peki o zaman, ben kendi tercihlerimi buraya yazarak başlıyorum - sonra herkesin fikriyle bir karara varabiliriz.

- Her zaman TAB
- `snake_case`
- pointer tanımlarken yıldız sağa
- Yorum yazarken /**/

Başka bir görüşünüz varsa lütfen belirtin ki dosyaya koyacak bir şeyimiz olsun. Ayrıca, ortada bir "evrensel stil" olduğunu hiç sanmıyorum, sırf bu 4 maddeyle en az 40 savaş açılabilir :)

**Reply by melikechan** [reply]
- CRLF kullanılması tercihim olur. Aslında tamamen LF kullanabiliriz ama \r\n kullanmaya alıştım C'de o yüzden CRLF tercihim.
Bence bu da eklenebilir.
